.github/workflows/build-test-docs.yml

```diff
@@ -0,0 +1,78 @@
+name: Build Test and Documentation
+# Build on all branches
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build-doc:
+    runs-on: ubuntu-latest
+    env:
+      PYTHON-VERSION: "3.8"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON-VERSION }}
+      - name: Upgrade pip
+        run: |
+          # install pip=>20.1 to use "pip cache dir"
+          python3 -m pip install --upgrade pip
+      - name: Get pip cache dir
+        id: pip-cache
+        run: echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-py${{ env.PYTHON-VERSION }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ env.PYTHON-VERSION }}
+
+      - name: Install dependencies
+        run: python3 -m pip install -r ./requirements.txt
+
+      - run: make docs -B
+
+  build-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.7", "3.8", "3.9", "3.10" ]
```

Comment:
do we still support 2.7?

Reply:
https://github.com/stravalib/stravalib/issues/231

Reply:
> Is it better to separate `doc yml` and `test yml`

Sorry for my confusion: is this an open question or suggestion for improvement? I don't consider myself an expert on Github Actions, so any advice is welcome.

> do we still support 2.7?

Good question, TBH I just copy-pasted the versions mentioned in the starter workflow. We may also need to add the recent 3.11?

Reply:
no need to add 3.11 for now I think.

Reply:
> Sorry for my confusion: is this an open question or suggestion for improvement? I don't consider myself an expert on Github Actions, so any advice is welcome.

its an open question.~

Reply:
hi! i think dropping python 2.x makes sense and we should do it. Most packages are doing this as it's end of life. i think generally supporting the latest several versions of python makes sense. we can always add 3.11 later as well. 

i'm going to look at the full build here in just a few but we should also setup a matrix to test on windows, mac and linux -

Reply:
so i think we should add a matrix  build. in this example we'd break the build into a docs only build that runs on linux or whatever os we want. Then a second that uses the matrix to run tests across versions of python and operating systems. 

Here is an example of what the matrix build would look like. i'm happy to help with this

https://github.com/earthlab/abc-classroom/blob/main/.github/workflows/run-tests-mac-win-linux.yml#L14
But we don't need in this example the various environments, we just need the os matrix part 
```
  setup-build:
    name: Ex1 (${{ matrix.python-version }}, ${{ matrix.os }})
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
        python-version: [3.7, 3.8, 3.9]

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python-version }}
```

Reply:
Ok, I'll create separate actions for test/lint/doc, where test is using a matrix as you mentioned above.